datalake/src/main/java/com/sequenceiq/datalake/service/sdx/flowcheck/CloudbreakFlowService.java

```diff
@@ -68,11 +68,40 @@ public FlowState getLastKnownFlowState(SdxCluster sdxCluster) {
             LOGGER.error("Flow chain id or resource {} not found in CB: {}, so there is no active flow!", sdxCluster.getClusterName(), e.getMessage());
             return FlowState.UNKNOWN;
         } catch (Exception e) {
-            LOGGER.error("Exception occured during checking if there is a flow for cluster {} in CB: {}", sdxCluster.getClusterName(), e.getMessage());
+            LOGGER.error("Exception occurred during checking if there is a flow for cluster {} in CB: {}", sdxCluster.getClusterName(), e.getMessage());
             return FlowState.UNKNOWN;
         }
     }
 
+    public FlowResponse getLastKnownFlowResponse(SdxCluster sdxCluster) {
+        FlowResponse flowResponse = new FlowResponse();
+        try {
+            if (sdxCluster.getLastCbFlowChainId() != null) {
+                LOGGER.info("Checking cloudbreak {} {} to get flow response", FlowType.FLOW_CHAIN, sdxCluster.getLastCbFlowChainId());
+                FlowCheckResponse flowCheckResponse = ThreadBasedUserCrnProvider.doAsInternalActor(
+                        regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                        () -> flowEndpoint.hasFlowRunningByChainId(sdxCluster.getLastCbFlowChainId()));
+                if (flowCheckResponse.getCurrentState() != null) {
+                    flowResponse.setCurrentState(flowCheckResponse.getCurrentState());
+                }
+                if (flowCheckResponse.getNextEvent() != null) {
+                    flowResponse.setTriggeredEvent(flowCheckResponse.getNextEvent());
+                }
+                if (flowCheckResponse.getFlowType() != null) {
+                    flowResponse.setFlowType(flowCheckResponse.getFlowType());
+                }
+            }
+            return flowResponse;
+        } catch (NotFoundException e) {
+            LOGGER.error("Flow chain id or resource {} not found in CB: {}, so there is no active flow!", sdxCluster.getClusterName(), e.getMessage());
+            return flowResponse;
+        } catch (Exception e) {
+            LOGGER.error("Exception occured during checking if there is a flow for cluster {} in CB: {}", sdxCluster.getClusterName(), e.getMessage());
+            return flowResponse;
+        }
```

Comment:
In this case, you are returning an object which is not initialized.

Reply:
Thank you @kkalvagadda1 , I updated the class to use flowCheckResponse and return null when exception happens. Let me know if it looks good!